version: "3.9"

services:
  db:
    build:
      context: database/
      dockerfile: Dockerfile
    container_name: database
    hostname: database
    ports:
      - "54329:5432"
    volumes:
      - ${PWD}/docker/volumes/postgres@15:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=gpb
      - POSTGRES_USER=gpb
      - POSTGRES_PASSWORD=gpb
      - "DB_URL=file:///tmp/sample.sql"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready" ]
      interval: 30s
      timeout: 30s
      retries: 3
    networks:
      - gpb_local

  import:
    build:
      context: import/
      dockerfile: docker/Dockerfile
    command: python manage.py import -f /tmp/sample/out.zip -z -t -d
    container_name: import
    hostname: import
    volumes:
      - ${PWD}/import:/app
      - ${PWD}/sample/:/tmp/sample/
    networks:
      - gpb_local
    depends_on:
      - db


networks:
  gpb_local:
